#!/usr/bin/env python
PACKAGE = "perception"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#int
gen.add("segmentation_radius",    int_t,    0, "An Integer parameter", 8,  0, 15)
gen.add("node_id",    int_t,    0, "An Integer parameter", 1,  0, 100)

#double
# gen.add("cluster_radius",    double_t,    0, "An Integer parameter", 1,  0, 5)

#bool
gen.add("bool_save",   bool_t,   0, "A Boolean parameter",  False)

#string
gen.add("save_name",    str_t,    0, "A string parameter",  "_node.pcd")

# size_enum = gen.enum([ gen.const("Low",      int_t, 0, "Low is 0"),
#                        gen.const("Medium",     int_t, 1, "Medium is 1"),
#                        gen.const("High",      int_t, 2, "Hight is 2")],
#                      "Select from the list")

# gen.add("size", int_t, 0, "Select from the list", 1, 0, 3, edit_method=size_enum)

exit(gen.generate(PACKAGE, "simuSegSave", "simuSegSave_"))
